packages/core/src/routines/CreateConfigRoutine.ts

```diff
@@ -118,6 +133,62 @@ export default class CreateConfigRoutine<Ctx extends ConfigContext> extends Rout
     return configPath;
   }
 
+  /**
+   * Create a configuration file for the defined driver based on a custom template
+   * provided by the consumer.
+   */
+  @Bind()
+  async createConfigFileFromTemplate(context: Ctx, configs: ConfigObject[]): Promise<Path> {
+    const { driver, tool } = this.options;
+    const { metadata } = driver;
+    const configPath = context.cwd.append(metadata.configName);
+    const templatePath = context.cwd.append(driver.options.template);
+    let template: ConfigTemplate;
+
+    if (!driver.options.template) {
+      throw new Error(tool.msg('errors:templateRequired'));
+    }
+
+    try {
+      template = requireModule(templatePath);
+    } catch {
+      throw new Error(
+        tool.msg('errors:templatePathMissing', {
+          path: templatePath.name(),
+        }),
+      );
+    }
+
+    if (typeof template !== 'function') {
+      throw new TypeError(
+        tool.msg('errors:templateTypeInvalid', {
+          type: typeof template,
+        }),
+      );
+    }
+
+    const config = template(configs, {
```

Comment:
Adding a type exported from beemo that describes this function and its arguments would be helpful